File: nestjs/templates/be-update.dto.ts
// Template: Update DTO
// Location: src/modules/<module-name>/dto/update-<module-name>.dto.ts
// All fields from CreateDto become optional automatically via PartialType.
// Using @nestjs/swagger PartialType (not @nestjs/mapped-types) so Swagger
// correctly shows all fields as optional in the generated docs.

import { PartialType } from '@nestjs/swagger'
import { CreateModuleNameDto } from './create-module-name.dto'

export class UpdateModuleNameDto extends PartialType(CreateModuleNameDto) {}

File: dotnet/templates/be-UpdateDto.cs
// Template: Update DTO
// Location: src/Modules/<ModuleName>/Dtos/Update<ModuleName>Dto.cs
// All fields are nullable to support partial updates (PATCH semantics).

using Swashbuckle.AspNetCore.Annotations;

namespace YourApp.Modules.ModuleName.Dtos;

public class UpdateModuleNameDto
{
    [SwaggerSchema("New name (optional)")]
    public string? Name { get; set; }

    [SwaggerSchema("New description (optional)")]
    public string? Description { get; set; }

    // TODO: mirror nullable versions of fields from CreateModuleNameDto
}
